### PR TITLE
Check for valid builder names before creating a build request

### PR DESCRIPTION
**Pull request by catlee**

### CODE REVIEW

File: bbb/services.py [hunk 1]
```diff
@@ -457,6 +463,20 @@ def _isAuthorized(self, buildername, scopes):
         else:
             return True
 
+    def _refreshAllowedBuilders(self):
+        now = arrow.now().timestamp
+        if self.allowed_builders is None or (now - self.allowed_builders_age) > 300:
+            try:
+                builders = requests.get(ALL_THE_THINGS_URL).json()['builders']
```
Comment (rail) [inline bbb/services.py]: Can you add `timeout=something` here? And maybe break it into multiple lines so you can use `raise_for_status()`?
Also what happens if the first attempt fails, and `self.allowed_builders` stays `None` for some time? Maybe set the initial value to `set()` and skip the check if it's empty?

Comment (catlee) [inline bbb/services.py]: I'm not sure what to do if this fails the first time. Ideally I think I'd like this to stop processing the message and try again later.

Comment (rail) [inline bbb/services.py]: I'm a bit concerned that if this happens we may end up rejecting new tasks without noticing.

Comment (rail) [inline bbb/services.py]: On the other hand the check (`x in y`) would fail, so we shouldn't be discarding messages.

**Pull request merged**